### PR TITLE
scan report v1

### PR DESCRIPTION
**Pull request by avrahams**
Create ScanResultReportV1 with ready to use Vulnerabilities and report summery

### CODE REVIEW

File: containerscan/rawdatastrucutres.go [hunk 1]
```diff
@@ -24,6 +24,20 @@ type ScanResultReport struct {
 	Session                  apis.SessionChain          `json:"session,omitempty"`
 }
 
+//ScanResultReportV1 replaces ScanResultReport
+type ScanResultReportV1 struct {
+	Designators     armotypes.PortalDesignator           `json:"designators"`
+	WLID            string                               `json:"wlid"`
+	ContainerName   string                               `json:"containerName"`
+	CustomerGUID    string                               `json:"customerGUID"`
```
Comment (LiorAlafiArmo) [inline containerscan/rawdatastrucutres.go]: לדעתי מיותר, השדה הזה גם ככה נמצא בdesignators

File: containerscan/rawdatastrucutres.go [hunk 2]
```diff
@@ -24,6 +24,20 @@ type ScanResultReport struct {
...
+	Timestamp       int64                                `json:"timestamp"`
+	ContainerScanID string                               `json:"containersScanID"`
+	Vulnerabilities []CommonContainerVulnerabilityResult `json:"vulnerabilities"`
+	Summery         *CommonContainerScanSummaryResult    `json:"summery,omitempty"`
+	PartNum         int                                  `json:"partNum"`
```
Comment (LiorAlafiArmo) [inline containerscan/rawdatastrucutres.go]: נא להכניס/למצוא pagination structure בתוך armotypes ולא כindividual fields, מחר אם נרצה להגדיל את זה הקונטקסט של השדות יהיה מפוזר...

File: containerscan/datastructuresmethods.go [hunk 3]
```diff
@@ -74,3 +74,32 @@ func (v *Vulnerability) IsRCE() bool {
 
 	return isRCE || strings.Contains(desc, "remote code execution") || strings.Contains(desc, "remote command execution") || strings.Contains(desc, "arbitrary code") || strings.Contains(desc, "code execution") || strings.Contains(desc, "code injection") || strings.Contains(desc, "command injection") || strings.Contains(desc, "inject arbitrary commands")
 }
+
+func (scanresult *ScanResultReportV1) Validate() bool {
```
Comment (LiorAlafiArmo) [inline containerscan/datastructuresmethods.go:78]: אולי כדאי לעטוף בinterface ולכתוב validator ו isRCE לאינטרפייס וזהו? (לא חובה-הצעה.)

File: containerscan/rawdatastrucutres.go (hunk 2 above)
Comment (avrahams) [reply containerscan/rawdatastrucutres.go]: @LiorAlafiArmo 
I see that PaginationMarks is in opa-utils
https://github.com/armosec/opa-utils/blob/master/reporthandling/v2/datastructures.go#L13
Does it make sene to create a dependency from cluster-container-scanner-api to opa-utils or PaginationMarks struct should move to another package?

File: containerscan/rawdatastrucutres.go (hunk 1 above)
Comment (avrahams) [reply containerscan/rawdatastrucutres.go]: This is the same implementation as ScanReportV0
It is used in `(scanresult *ScanResultReport) GetDesignatorsNContext() (*armotypes.PortalDesignator, []armotypes.ArmoContext) {`

File: containerscan/datastructuresmethods.go [hunk 4]
```diff
@@ -74,3 +74,32 @@ func (v *Vulnerability) IsRCE() bool {
...
+		return false
+	}
+
+	if _, err := uuid.Parse(scanresult.CustomerGUID); err != nil {
+		return false
+	}
+	return true
+}
+
+func (scanresult *ScanResultReportV1) GetDesignatorsNContext() (*armotypes.PortalDesignator, []armotypes.ArmoContext) {
```
Comment (Bezbran) [inline containerscan/datastructuresmethods.go]: add unitests

File: containerscan/rawdatastrucutres.go [hunk 5]
```diff
@@ -24,6 +24,20 @@ type ScanResultReport struct {
...
+//ScanResultReportV1 replaces ScanResultReport
+type ScanResultReportV1 struct {
+	Designators     armotypes.PortalDesignator           `json:"designators"`
+	WLID            string                               `json:"wlid"`
+	ContainerName   string                               `json:"containerName"`
+	CustomerGUID    string                               `json:"customerGUID"`
+	Timestamp       int64                                `json:"timestamp"`
+	ContainerScanID string                               `json:"containersScanID"`
+	Vulnerabilities []CommonContainerVulnerabilityResult `json:"vulnerabilities"`
+	Summery         *CommonContainerScanSummaryResult    `json:"summery,omitempty"`
```
Comment (Bezbran) [inline containerscan/rawdatastrucutres.go]: Summary

File: containerscan/rawdatastrucutres.go (hunk 1 above)
Comment (Bezbran) [reply containerscan/rawdatastrucutres.go]: In addition - for registry scan we have no WLID nor container name

Comment (Bezbran) [reply containerscan/rawdatastrucutres.go]: we just need to see if UI is using this field - if so we should tell them to stop using it or calculate it only upon Elastic insertion

**Review by Bezbran — CHANGES_REQUESTED** [review]

File: containerscan/datastructuresmethods.go (hunk 3 above)
Comment (avrahams) [reply containerscan/datastructuresmethods.go:78]: Bez input was not to put effort in V0 scan reports

File: containerscan/rawdatastrucutres.go (hunk 1 above)
Comment (avrahams) [reply containerscan/rawdatastrucutres.go]: done

**Review by Bezbran — APPROVED** [review]

**Pull request merged**